compiler/quilt/tools/const.py

```diff
@@ -70,3 +70,21 @@ class TargetType(Enum):
 # Exit codes
 EXIT_KB_INTERRUPT = 4
 TEAM_ID_ERROR = "Invalid team name: "
+
+
+# Exceptions
+class QuiltException(Exception):
+    """Base Exception for Quilt
+
+    Any time an exception is raised by Quilt code and isn't caught by
+    the module raising it, it should use a `QuiltException` or subclass.
+
+    `QuiltException` is the base for all quilt exceptions, such as
+    `CommandException`, `BuildException`, etc.
+
+    All `QuiltException` instances should have a user friendly message,
+    even if not expected to be seen by a user.
+    """
+    def __init__(self, message, *args):
+        super(QuiltException, self).__init__(message, *args)
```

Comment:
Looks like you don't need to call `super`; examples in Python docs don't show it, and you can't really do anything useful with the parent class. (`Exception` has a `message` attribute, but it's deprecated; not sure what happens to `*args` - might as well get rid of those.)

Reply:
There's a whole small topic around this that I won't go into, that relates to language consistency vs convenience, and Python3's choices regarding usage of parameters by a function they weren't passed to -- however, again, I'll skip all the cruft about features that may or may not stay.  I am, however, happy to have that conversation if you want, it's kinda interesting.  ..but, noise.

What relates directly to us is this:
```python
# Without super call, python2:
str(QuiltException('Error message')) -> ""
# With super call, python2:
str(QuiltException('Error message')) -> "Error message"

# With or without super call, python3:
str(QuiltException('Error message')) -> "Error message"
```

We use `"Some Prefix: " + str(error)` fairly frequently.

There's also a similar issue that this has caused me to notice, though, too -- if you rely on Python3's 'convenience' of (effectively -- I don't know what the C code does, haven't looked at it) calling super() for you when you subclass an exception, it always uses all arguments, which may not be (Read:  Is not, in our case) ideal, because passing more than one argument to `QuiltException` would result in a string repr of the tuple of arguments (instead of a string of the message).

..so:
I'll change that `super` call to only include the 'message' argument.

Reply:
Haha, ok, sounds good.